exporter/exporterhelper/common.go

```diff
@@ -265,28 +260,42 @@ func newBaseExporter(set exporter.Settings, signal component.DataType, osf obsre
 		return nil, err
 	}
 
-	if be.batcherCfg.Enabled {
-		bs := newBatchSender(be.batcherCfg, be.set, be.batchMergeFunc, be.batchMergeSplitfunc)
-		for _, opt := range be.batcherOpts {
-			err = multierr.Append(err, opt(bs))
-		}
-		if bs.mergeFunc == nil || bs.mergeSplitFunc == nil {
-			err = multierr.Append(err, fmt.Errorf("WithRequestBatchFuncs must be provided for the batcher applied to the request-based exporters"))
-		}
-		be.batchSender = bs
+	// Set up queue sender if queue is enabled
+	if be.marshaler != nil && be.unmarshaler != nil && be.queueCfg.Enabled {
+		q := be.queueFactory(context.Background(), exporterqueue.Settings{
+			DataType:         be.signal,
+			ExporterSettings: be.set,
+		}, exporterqueue.Config{
+			Enabled:      be.queueCfg.Enabled,
+			NumConsumers: be.queueCfg.NumConsumers,
+			QueueSize:    be.queueCfg.QueueSize,
+		})
+		be.queueSender = newQueueSender(q, be.set, be.queueCfg.NumConsumers, be.exportFailureMessage, be.obsrep)
 	}
 
-	if be.queueCfg.Enabled {
+	if be.marshaler == nil && be.unmarshaler == nil && be.exporterQueueCfg.Enabled {
 		set := exporterqueue.Settings{
 			DataType:         be.signal,
 			ExporterSettings: be.set,
 		}
-		be.queueSender = newQueueSender(be.queueFactory(context.Background(), set, be.queueCfg), be.set, be.queueCfg.NumConsumers, be.exportFailureMessage, be.obsrep)
+		be.queueSender = newQueueSender(be.queueFactory(context.Background(), set, be.exporterQueueCfg), be.set, be.exporterQueueCfg.NumConsumers, be.exportFailureMessage, be.obsrep)
 		for _, op := range options {
 			err = multierr.Append(err, op(be))
 		}
 	}
 
+	// Set up batch sender if batching is enabled
```

Comment:
Moving batch sender initialization down because in the future we will only initialize batch sender if queue sender is not initialized.